Advanced-Recruit-Training/src/BatteryStateOfChargeService.cpp

```diff
@@ -16,23 +24,48 @@ BatteryStateOfChargeService::~BatteryStateOfChargeService()
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+
+    return ampHour_;
 }
 
 bool BatteryStateOfChargeService::isCharging() const
 {
+
+    if(current_<0){
+
+        return true;
+    }
     return false;
 }
 
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
-{
-    return QTime::currentTime();
+{    QTime charging;
+     charging.setHMS(0,0,0,0);
+
+    if(isCharging()==true){
+    charging= charging.addMSecs(qAbs(ampHour_/current_*3600000));
```

Comment:
3600000 this is a magic number, define a constant in the namespace above and try to avoid them throughout your program as much as possible